### PR TITLE
WDIP-16: enable hotkeys for selecting flyaway menu options

### PR DESCRIPTION
**Pull request by utiq**
Re-openning #115

### CODE REVIEW

**Comment by adamlerer** [comment]
I think there's still a problem here: the hotkeys need to differentiate move vs "move via" since sometimes both are possible (and you can't just default to "move via").

File: botgamecreate.php [hunk 1]
```diff
@@ -101,7 +101,7 @@
 		$botNum = $countryCount - 1;
 		// $tabl = $DB->sql_tabl("SELECT id FROM wD_Users WHERE type LIKE '%bot%' LIMIT ".$botNum);
 		// Use a specific bot for a specific variant for now. A new Config:: function is needed to be able to flexibly map variants to certain specialized bots
-		$tabl = $DB->sql_tabl("SELECT id FROM wD_Users WHERE type LIKE '%bot%' ".($input['variantID']==15?" AND username='FairBot' ":"")." LIMIT ".$botNum);
+		// $tabl = $DB->sql_tabl("SELECT id FROM wD_Users WHERE type LIKE '%bot%' ".($input['variantID']==15?" AND username='FairBot' ":"")." LIMIT ".$botNum);
```
Comment (adamlerer) [inline botgamecreate.php:104]: Why did you comment this out?

File: beta-src/src/components/map/components/WDFlyoutContainer.tsx [hunk 2]
```diff
@@ -19,80 +17,113 @@ interface WDFlyoutContainerProps {
...
+    }
+  };
+
+  useKeyPressEvent("h", () => clickHandler("Hold"));
+  useKeyPressEvent("d", () => clickHandler("Hold"));
+  useKeyPressEvent("m", moveAction);
+  useKeyPressEvent("a", moveAction);
+  useKeyPressEvent("v", moveAction);
+  useKeyPressEvent("s", () => clickHandler("Support"));
+  useKeyPressEvent("c", fleetAndSeaAction);
```
Comment (adamlerer) [inline beta-src/src/components/map/components/WDFlyoutContainer.tsx:78]: Here's explicitly what I think this should look like to give correct behavior for vias.

```suggestion
  const canVia: boolean =
    unit?.unit?.type === "Army" &&
    legalOrders.legalViasByUnitID[order.unitID].length > 0;
  };

  useKeyPressEvent("h", () => clickHandler("Hold"));
  useKeyPressEvent("d", () => clickHandler("Hold"));
  useKeyPressEvent("m", () => clickHandler("Move"));
  useKeyPressEvent("a", () => clickHandler("Move"));
  useKeyPressEvent("v", () => canVia && clickHandler("Move", "Yes"));
  useKeyPressEvent("s", () => clickHandler("Support"));
  useKeyPressEvent("c", () => clickHandler("Convoy"));
```

**Review by adamlerer — CHANGES_REQUESTED** [review]
I heard requests again from the players for this feature, so lets get it merged.

**Comment by utiq** [comment]
> I heard requests again from the players for this feature, so lets get it merged.

sounds good, I'll fix the changes requests

**Comment by utiq** [comment]
@adamlerer I fixed the conflicts and I committed the suggestion you made. Merged.

**Pull request merged**